Documentation/network/concepts/ipam/index.rst

```diff
@@ -26,6 +26,10 @@ Multiple CIDRs per node      ❌                     ❌                       
 Dynamic CIDR/IP allocation   ❌                     ❌                       ✅          ✅          ✅              ✅                ❌
 ============================ ====================== ======================== =========== =========== =============== ================= ===============
 
+Changing the IPAM mode of an existing cluster is impossible.
+You must reinstall Cilium to change the IPAM mode.
+Before reinstalling Cilium, delete all CiliumNode CRD objects.
```

Comment:
Here's another alternative proposal:
```suggestion
Do not change the IPAM mode of an existing cluster. Changing the IPAM mode in
a live environment may cause persistent connectivity disruption for existing workloads.
The safest path to change IPAM mode is to install a fresh Kubernetes cluster with the new IPAM configuration.
```

I feel this would make it clear that the Cilium community has not attempted to solve this problem yet, and if we want to solve this problem then we can open issues/CFPs/PRs and work through the process to provide a path.

Reply:
Is a fresh cluster needed? I guess removing Cilium and run https://docs.cilium.io/en/stable/cmdref/cilium_cleanup/ will do the trick?

Reply:
Not quite, mainly because that cilium cleanup only cleans local node state, not state in k8s, so the step that was referenced earlier about deleting CiliumNode resources won't happen AFAIK.

Reply:
Yes, that's also my understanding. I'm refering to "install a fresh cluster" So in theory deleting CiliumNode CRD + run the cleanup.

Reply:
Ah yes that should technically work as well. We could alternatively describe it as completely removing Cilium from the cluster including all k8s resources and then freshly installing Cilium.

Reply:
Yes, I think that's the best approach. Then we can link from "changing IPAM mode" to this section. I could work on that next week. Maybe you can give me some hints on where I should add that.

Reply:
Ah, what I had in mind was just to tersely describe here that Cilium must be completely removed and reinstalled. But yeah it seems reasonable to document that too. I'm not seeing an obvious place in the hierarchy for this right now, maybe @zacharysarah might have a thought on this?

Reply:
Would it make sense to add that heading instead under troubleshooting? Minor tweaks to @joestringer's proposed update:
```suggestion
Don't change the IPAM mode of an existing cluster. Changing the IPAM mode in
a live environment may cause persistent disruption of connectivity for existing workloads.
The safest path to change IPAM mode is to install a fresh Kubernetes cluster with the new IPAM configuration.
```

> I feel this would make it clear that the Cilium community has not attempted to solve this problem yet, and if we want to solve this problem then we can open issues/CFPs/PRs and work through the process to provide a path.

Would adding explicit language to that effect be useful, or is the statement sufficiently self-disclosing in context? (I think your language here would work great, with some tweaks.)

Reply:
implemented the recommendations

Reply:
> > I feel this would make it clear that the Cilium community has not attempted to solve this problem yet, and if we want to solve this problem then we can open issues/CFPs/PRs and work through the process to provide a path.
>
> Would adding explicit language to that effect be useful, or is the statement sufficiently self-disclosing in context? (I think your language here would work great, with some tweaks.)

I think that it would help to call this out explicitly. Even though it's Open Source and hence there's always an opportunity to change the way things work, I think that it's often difficult for users to understand whether something is impossible by design or just that no-one solved it yet. I'll propose some wording to this effect.